src/main/java/htsjdk/tribble/index/IndexFactory.java

```diff
@@ -162,20 +162,32 @@ public static IndexType getIndexType(final BufferedInputStream is) {
      * @param indexFile from which to load the index
      */
     public static Index loadIndex(final String indexFile) {
+        return loadIndex(indexFile, null);
+    }
+
+    /**
+     * Load in index from the specified file.   The type of index (LinearIndex or IntervalTreeIndex) is determined
+     * at run time by reading the type flag in the file.
+     *
+     * @param indexFile from which to load the index
+     */
+    public static Index loadIndex(final String indexFile, Function<SeekableByteChannel, SeekableByteChannel> indexWrapper) {
         // Must be buffered, because getIndexType uses mark and reset
-        try (BufferedInputStream bufferedInputStream = new BufferedInputStream(indexFileInputStream(indexFile), Defaults.NON_ZERO_BUFFER_SIZE)){
+        try (BufferedInputStream bufferedInputStream = new BufferedInputStream(indexFileInputStream(indexFile, indexWrapper), Defaults.NON_ZERO_BUFFER_SIZE)) {
             final Class<Index> indexClass = IndexType.getIndexType(bufferedInputStream).getIndexType();
             final Constructor<Index> ctor = indexClass.getConstructor(InputStream.class);
             return ctor.newInstance(bufferedInputStream);
+        } catch (final TribbleException ex) {
+            throw ex;
```

Comment:
There was previously re-wrapping tribble exceptions as runtime exceptions.  This makes it more consistent and specific.  Since a TribbleException is a RuntimeException, this is backwards compatible.